.github/workflows/release.yml

```diff
@@ -0,0 +1,135 @@
+# SPDX-FileCopyrightText: 2023 Forschungszentrum Jülich GmbH (FZJ)
+#
+# SPDX-License-Identifier: CC0-1.0
+
+# SPDX-FileContributor: Oliver Bertuch
+
+name: Publish Python 🐍 distribution 📦 to PyPI and TestPyPI
+
+on:
+  push:
+    branches:
+      - main
+      - develop
+    tags:
+      - 'v**'
+
+jobs:
+  build:
+    name: Build distribution 📦
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v4
+    - name: Set up Python
+      uses: actions/setup-python@v4
+      with:
+        python-version: "3.10"
+    - name: Install pypa/build
+      run: >-
+        python3 -m
+        pip install
+        build toml-cli
+        --user
+    - name: Reset package name and version for dev snaps
+      if: github.ref_name == 'develop'
+      run: |
+        toml set --toml-path pyproject.toml tool.poetry.name hermes-test
+        VERSION=$(toml get --toml-path pyproject.toml tool.poetry.version)
+        toml set --toml-path pyproject.toml tool.poetry.version "$VERSION.dev${{ github.run_number }}"
+    - name: Build a binary wheel and a source tarball
+      run: python3 -m build
+    - name: Store the distribution packages
+      uses: actions/upload-artifact@v3
+      with:
+        name: python-package-distributions
+        path: dist/
+
+  publish-to-pypi:
+    name: >-
+      Publish Python 🐍 distribution 📦 to PyPI
+    if: startsWith(github.ref, 'refs/tags/')  # only publish to PyPI on tag pushes
+    needs:
+      - build
+    runs-on: ubuntu-latest
+    environment:
+      name: pypi
+      url: https://pypi.org/p/hermes  # Replace <package-name> with your PyPI project name
+    permissions:
+      id-token: write  # IMPORTANT: mandatory for trusted publishing
+
+    steps:
+    - name: Download all the dists
+      uses: actions/download-artifact@v3
+      with:
+        name: python-package-distributions
+        path: dist/
+    - name: Publish distribution 📦 to PyPI
+      uses: pypa/gh-action-pypi-publish@release/v1
```

Comment:
I can't test this without a tag, which I don't want to create (yet). Maybe we can agree that after merging #236 we create a release candidate tag first to have some leeway for testing this flow?